lib/data_update_scripts/20210512033821_clean_up_site_config.rb

```diff
@@ -0,0 +1,55 @@
+module DataUpdateScripts
+  class CleanUpSiteConfig
+    OBSOLETE_FIELDS = %w[
+      allowed_registration_email_domains
+      authentication_providers
+      campaign_articles_expiry_time
+      campaign_articles_require_approval
+      campaign_call_to_action
+      campaign_featured_tags
+      campaign_hero_html_variant_name
+      campaign_sidebar_enabled
+      campaign_sidebar_image
+      campaign_url
+      collective_noun
+      collective_noun_disabled
+      community_copyright_start_year
+      community_description
+      community_emoji
+      community_member_label
+      community_name
+      default_font
+      experience_high
+      experience_low
+      feed_strategy
+      feed_style
+      home_feed_minimum_score
+      primary_brand_color_hex
+      public
+      rate_limit_article_update
+      rate_limit_comment_antispam_creation
+      rate_limit_comment_creation
+      rate_limit_email_recipient
+      rate_limit_feedback_message_creation
+      rate_limit_follow_count_daily
+      rate_limit_image_upload
+      rate_limit_listing_creation
+      rate_limit_organization_creation
+      rate_limit_published_article_antispam_creation
+      rate_limit_published_article_creation
+      rate_limit_reaction_creation
+      rate_limit_send_email_confirmation
+      rate_limit_user_subscription_creation
+      rate_limit_user_update
+      spam_trigger_terms
+      staff_user_id
+      tag_feed_minimum_score
+      tagline
+      user_considered_new_days
+    ].freeze
+
+    def run
+      SiteConfig.delete_by(var: OBSOLETE_FIELDS)
```

Comment:
This is okay now because the data updates moving these values _already_ happened *AND* nothing was writing to these fields since we moved/copied the values?

Reply:
Totally *non-blocking* and I'm likely creating complications out of thin air, but the idea that we know whether and when people have upgraded code has a shelf life.

While we probably still control _most_ of the supported forem instances, is there any real risk to a self-hosting pioneer trying to update in the future (someone who had started a Forem last week while SiteConfig was still a location for much of the configuration, and attempts to update to the next release after this is merged and released)? 

Is the worst case scenario that an update removes configuration that _didn't_ also already get copied elsewhere? Or is this (because of in-order execution) guaranteed to run _after_ the other update scripts. Is this some chasm where standing on one side you have a working instance, and updating to the other side where scripts have been deleted from the repo/filesystem and thus will _never_ run you have a non-functioning system, or is it only a crack where the outcome is a poorly configured system (you can still login to admin and set things up, but you might have lost some of your config choices)?

Reply:
> This is okay now because the data updates moving these values already happened AND nothing was writing to these fields since we moved/copied the values?

@djuber I may be missing a case or two, but generally, updates of values always went to the right tables (the `settings_*` ones) though we still showed information from `SiteConfig` in several places which means updates wouldn't have been properly reflected. 

> is there any real risk to a self-hosting pioneer trying to update in the future (someone who had started a Forem last week while SiteConfig was still a location for much of the configuration, and attempts to update to the next release after this is merged and released)?

Not if they ran all data update scripts.

> Or is this (because of in-order execution) guaranteed to run after the other update scripts. 

I wanna say yes, but we are talking about software so I'll say "to a reasonable degree of certainty" instead.

>  Is this some chasm where standing on one side you have a working instance, and updating to the other side where scripts have been deleted from the repo/filesystem and thus will never run you have a non-functioning system

See above. Lately, when we deleted scripts we tried to also always remove the script itself (`DataUpdateScript.delete_bu(name: "...")`) so hopefully we are good on that front.

Thanks for bringing up these points, they are are very valuable and we should talk about these things more. In this specific instance I will take a leap of faith and hope for the best because I really need to unblock myself to be able to finish the work on this RFC. 🤞